packages/components/src/components/Snackbar/useSnackbar.tsx

```diff
@@ -15,23 +16,36 @@ export function useSnackbar() {
   const toast = Toast.useToast()
   const context = useContext(SnackbarContext)
   const defaultOffset = useSnackbarDefaultOffset()
+  const screenReaderEnabled = useIsScreenReaderEnabled()
 
   if (!context) {
     throw new Error('useSnackbar must be used within a SnackbarProvider')
   }
 
-  const { offset, setOffset } = context
-
   const show = (message: string, snackbarOptions?: SnackbarOptions) => {
-    toast.hideAll()
+    const { offset, setOffset, duration, setDuration } = context
 
-    if (snackbarOptions?.offset) {
-      setOffset(snackbarOptions.offset)
-    } else if (offset !== defaultOffset) {
-      setOffset(defaultOffset)
+    // Adjust offset if provided or reset to default
+    const newOffset = snackbarOptions?.offset || defaultOffset
+
+    // Only call setOffset if different from current to avoid re-render
+    if (newOffset !== offset) {
+      setOffset(newOffset)
     }
 
-    return toast.show(message, { data: snackbarOptions })
+    // Auto-dismiss if screen reader is on and there is no action button
+    const newDuration =
+      screenReaderEnabled && !snackbarOptions?.onActionPressed
+        ? SNACKBAR_DURATIONS.SCREEN_READER
+        : SNACKBAR_DURATIONS.DEFAULT
+
+    // Only call setDuration if different from current to avoid re-render
+    if (newDuration !== duration) {
+      setDuration(newDuration)
+    }
+
+    toast.hideAll()
+    toast.show(message, { data: snackbarOptions })
```

Comment:
Did it not work to send the duration at show time? There's a prop for it with the library that I assumed would override the default:
```suggestion
    toast.show(message, { data: snackbarOptions, duration: newDuration })
```
would allow meaningful cleanup if it works as expected to not need to involve the context.

Reply:
Nice. Must have missed this in the toast props. Good catch. Replaced.